### PR TITLE
Runtime fixes

### PR DESCRIPTION
**Pull request by DTraitor**
### Purpose
Fixes some runtimes

### CODE REVIEW

File: code/game/machinery/ds13/bpl/biomass_storage.dm [hunk 1]
```diff
@@ -9,6 +9,13 @@
 	density = TRUE
 	anchored = TRUE
 
+	var/obj/machinery/recycling_tank/recycle
+
+/obj/structure/reagent_dispensers/biomass/Destroy()
+	recycle.storage = null
```
Comment (NanakoAC) [inline code/game/machinery/ds13/bpl/biomass_storage.dm]: check to be sure the storage == src before nulling it

File: code/modules/clothing/spacesuits/rig/rig_verbs.dm [hunk 2]
```diff
@@ -16,22 +16,22 @@
...
-	if(!check_power_cost(usr, visor.use_power_cost, visor.active_power_cost, 0, visor))
+	if(!visor)
+		to_chat(usr, "<span class='warning'>The RIG does not have a configurable visor.</span>")
 		return
 
 	if(canremove)
 		to_chat(usr, "<span class='warning'>The suit is not active.</span>")
 		return
 
+	if(!istype(wearer) || !wearer.back == src)
```
Comment (NanakoAC) [inline code/modules/clothing/spacesuits/rig/rig_verbs.dm]: don't check back, check wearing_rig var on human

File: code/modules/mob/living/carbon/human/species/necromorph/twitcher.dm [hunk 3]
```diff
@@ -170,6 +170,9 @@ All of these properties combined make Step Strike tricky and disorienting to use
 			var/vector2/delta = get_new_vector(A.x - x, A.y - y)
 			delta = delta.ToMagnitude(3)
 			var/turf/blink_target = locate(x+delta.x, y+delta.y, z)
+			if(!check_trajectory(blink_target, src, pass_flags))
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/species/necromorph/twitcher.dm]: if something blocks then the trajectory should be scaled back to a smaller magnitude until it doesn't collide, run a loop for that reducing it by one each time

File: code/modules/mob/living/carbon/taste.dm [hunk 4]
```diff
@@ -20,7 +20,7 @@ calculate text size per text.
 	var/minimum_percent = 15
 	if(ishuman(taster))
 		var/mob/living/carbon/human/H = taster
-		minimum_percent = round(15/ (H.isSynthetic() ? TASTE_DULL : H.species.taste_sensitivity))
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/taste.dm]: I dont think this is correct, you probably wouldn't use round on a hardcoded number

File: code/modules/multiz/structures.dm [hunk 5]
```diff
@@ -109,9 +109,9 @@
 
 	M.visible_message("<span class='notice'>\The [M] begins climbing [direction] \the [src]!</span>",
 	"You begin climbing [direction] \the [src]!",
-	"You hear the grunting and clanging of a metal ladder being used.")
+	"You hear the grunting and clanging of a metal ladder being used.", 5)
```
Comment (NanakoAC) [inline code/modules/multiz/structures.dm]: whats this 5 for?

File: code/modules/nano/nanoui.dm [hunk 6]
```diff
@@ -417,7 +417,7 @@ nanoui is used to open and update nano browser uis
   * @return nothing
   */
 /datum/nanoui/proc/open()
-	if(!user.client)
+	if(!user || !user.client)
```
Comment (NanakoAC) [inline code/modules/nano/nanoui.dm]: do `if (!user?.client)`

**Review by NanakoAC — CHANGES_REQUESTED** [review]
some tweaks needed

File: code/modules/multiz/structures.dm (hunk 5 above)
Comment (DTraitor) [reply code/modules/multiz/structures.dm]: range, it was null

File: code/modules/clothing/spacesuits/rig/rig_verbs.dm [hunk 7]
```diff
@@ -6,7 +6,7 @@
 	set category = "RIG"
 	set src = usr.contents
 
-	if(wearer && wearer.back == src)
+	if(!wearer.wearing_rig == src)
```
Comment (NanakoAC) [inline code/modules/clothing/spacesuits/rig/rig_verbs.dm]: wearer?.wearing_rig

File: code/modules/clothing/spacesuits/rig/rig_verbs.dm [hunk 8]
```diff
@@ -59,7 +59,7 @@
 	set category = "RIG"
 	set src = usr.contents
 
-	if(!istype(wearer) || !wearer.back == src)
+	if(!wearer.wearing_rig == src)
```
Comment (NanakoAC) [inline code/modules/clothing/spacesuits/rig/rig_verbs.dm]: same here

File: code/modules/clothing/spacesuits/rig/rig_verbs.dm [hunk 9]
```diff
@@ -91,7 +91,7 @@
...
```
Comment (NanakoAC) [inline code/modules/clothing/spacesuits/rig/rig_verbs.dm]: and likewise, all instances of this

File: code/modules/mob/living/abilities/step_strike.dm [hunk 10]
```diff
@@ -77,6 +77,9 @@
 
 		target_loc = pick(candidates)
 
+		if(!check_trajectory(target_loc, user, user.pass_flags))
```
Comment (NanakoAC) [inline code/modules/mob/living/abilities/step_strike.dm]: if the target location isnt reachable then pick_n_take another, don't just randomly fail

File: code/modules/mob/living/carbon/human/species/necromorph/twitcher.dm [hunk 11]
```diff
@@ -170,6 +170,23 @@ All of these properties combined make Step Strike tricky and disorienting to use
 			var/vector2/delta = get_new_vector(A.x - x, A.y - y)
 			delta = delta.ToMagnitude(3)
 			var/turf/blink_target = locate(x+delta.x, y+delta.y, z)
+			while(blink_target)
+				if(!check_trajectory(blink_target, src, pass_flags))
+					if(delta.x != 0 || delta.y != 0)
```
Comment (NanakoAC) [inline code/modules/mob/living/carbon/human/species/necromorph/twitcher.dm]: this isn't how you scale vectors, use ToMagnitude

**Review by NanakoAC — CHANGES_REQUESTED** [review]
more stuff needed

File: code/modules/mob/living/abilities/step_strike.dm [hunk 12]
```diff
@@ -77,8 +77,16 @@
 
 		target_loc = pick(candidates)
 
-		if(!check_trajectory(target_loc, user, user.pass_flags))
-			return	//An obstacle prevents you from step striking. There is no reason to check the closest possible place to teleport since step strike can be used only if target is 2 tiles away or closer
+		var/sfat = TRUE //search_for_avalaibe_turfs
+		while(sfat)
+			if(!check_trajectory(target_loc, user, user.pass_flags|PASS_FLAG_TABLE|PASS_FLAG_NOMOB))
+				if(!candidates.len)
+					return //An obstacle prevents you from step striking. There is no reason to check the closest possible place to teleport since step strike can be used only if target is 2 tiles away or closer
```
Comment (NanakoAC) [inline code/modules/mob/living/abilities/step_strike.dm:84]: you've got this mixed up, its the charge that requires a minimum distance, step strike can be used to dance around someone you're already in melee with

**Review by NanakoAC — CHANGES_REQUESTED** [review]
step strike and charge code is mixed up

File: code/modules/mob/living/abilities/step_strike.dm (hunk 12 above)
Comment (DTraitor) [reply code/modules/mob/living/abilities/step_strike.dm:84]: it works when the person is 2 tiles away and if there is window between you and your target you'll phase through it without this check

**Review by NanakoAC — APPROVED** [review]

**Pull request merged**